lib/compliance/blackduck.js

```diff
@@ -0,0 +1,244 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const Q = require('q')
+const _ = require('lodash')
+const nlf = require('nlf')
+const logger = require('../logger')
+
+module.exports = {
+
+  /** *****************************************************************
+   *  External methods are defined in object so they can be stubbed  *
+   *******************************************************************/
+
+  stat: Q.denodeify(fs.stat),
+  mkdir: Q.denodeify(fs.mkdir),
+  writeFile: Q.denodeify(fs.writeFile),
+  readFile: Q.denodeify(fs.readFile),
+  findLicenses: Q.denodeify(nlf.find),
```

Comment:
We're using `rewire` in the test, so these don't need to be defined in the object, they just can't be `const`, so at the module level, you can have:

```js
let stat = Q.denodeify(fs.stat)
let mkdir = Q.denodeify(fs.mkdir)
let writeFile = Q.denodeify(fs.writeFile)
let readFile = Q.denodeify(fs.readFile)
let findLicenses = Q.denodeify(nlf.find)
```

Or, better yet, using `const Promise = require('promise')`:
```js
let stat = Promise.denodeify(fs.stat)
let mkdir = Promise.denodeify(fs.mkdir)
let writeFile = Promise.denodeify(fs.writeFile)
let readFile = Promise.denodeify(fs.readFile)
let findLicenses = Promise.denodeify(nlf.find)
```